CHANGELOG.md

```diff
@@ -2,10 +2,11 @@
 
 ## WIP
 
+- Add support for Micro (via @icopp)
```

Comment:
A few changes in this PR have nothing to do with Franz, let's isolate relevant changes together, and make other PRs for other changes.

thanks!